content/articles/statefull-vs-stateless-architectures/index.md

```diff
@@ -0,0 +1,116 @@
+
+RESTful services developed in popularity as a result of its stateless design, eventually surpassing SOAP in usage. In the design world, Graph QL is gaining traction. RESTful services, like Graph QL-based services, are stateless thus e nd-users have a lot of influence over what data is sent to and from the servers.
+
+### Differences Between Atateless and Atateful Architectures
+The main difference between the two architectures is how they handle user states and sessions. The differences between the two areas are highlighted in the table below:
+
+**Stateless**|**Stateful**
+:-----:|:-----:
+Rely on clients to store session state information using cookies and local storage.|Rely on servers to store session state information.
+Server design is simple as it does not handle most of the functionality.|Server design is complex as it handles a lot of complexity
+The server handles requests much faster as it does not need to look beyond the single request sent to it as the request contains all the information the server needs to handle it.|The server handles user requests relatively slower compared to stateless architectures servers. The server has to manage the state sessions all on its own, thus, a response gets to the client much slower.
+The server and client are independent and loosely coupled.|The server and client are dependent on each other.
+Servers in this architecture are crash-proof. They can be restarted after a failure. The other servers could continue to handle a user request when the server handling it goes down.|Servers in this architecture do not recover easily from a crash. They store session status and information about a client during sessions. So if they crash, the session has to start over.
+
+
+### Advantages Stateless Architecture
+1.	A network's monitoring system does not need to read the information beyond a single request to comprehend it, which improves visibility. This is because, every request is handled in isolation.
+2.	They simplify the design of the server since most functionality is handled client-side.
+3.	It is very reliable as it recovers from partial failures easily. The server does not need to store any particular state or session; thus, a failed server can restart or transfer the client's request to another server of the same organization.
+4.	Scalability is achieved easily. Since the servers do not have to store session states, resources can be freed to handle other incoming requests.
+5.	They can easily handle multiple sessions simultaneously since every request is isolated.
+6.	Memory usage on the server-side is reduced. This is because the information used to handle requests is not stored there.
+
+### Disadvantages of Stateless Protocols 
```

Comment:
The Disadvantage of Stateless Protocols. 
You've only provided one disadvantage, so it cannot be disadvantages. 

Also this 'Advantages of Stateful Architecture'